workfloworchestrator/src/main/java/com/greenops/workfloworchestrator/ingest/handling/EventHandlerImpl.java

```diff
@@ -97,6 +96,9 @@ public void handleEvent(Event event) {
         var pipelineData = fetchPipelineData(event, gitRepoUrl, gitCommit);
         if (pipelineData == null) throw new AtlasNonRetryableError("The pipeline doesn't exist");
 
+        if (isDuplicateEvent(event)) {
+            return;
+        }
```

Comment:
in the handleStepTriggerEvent method, if the steptrigger is a rollback, clear the event key cache